packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx

```diff
@@ -65,6 +93,21 @@ export class TooltipHost extends BaseComponent<ITooltipHostProps, ITooltipHostSt
     );
   }
 
+  /** Check whether the host's content is overflowing */
+  private _checkForOverflow() {
+    const { onlyShowIfOverflow, overflowElement } = this.props;
+    if (onlyShowIfOverflow) {
+      const { isOverflowing } = this.state;
+      const isNowOverflowing = overflowElement && isElementOverflowing(overflowElement);
+
+      if (isNowOverflowing !== isOverflowing) {
+        this.setState({
+          isOverflowing: isNowOverflowing
+        });
```

Comment:
Any reason why we are keeping "isOverflowing" as a state? how about checking for overflow only on mouseEnter? Seems like we are doing a lot of computation during react lifecycle which may be triggered by parent's state change and produce same result for isOverflowing.